File: C#api/Tests/test_DELETE_handler.cs
using System.Net.Http;
using System.Net;
using System.Text;
using System.Threading.Tasks;
using Models;
using Newtonsoft.Json;
using Xunit;

public class ApiDeleteTests
{
    private readonly HttpClient _client;

    public ApiDeleteTests()
    {
        _client = new HttpClient { BaseAddress = new Uri("http://localhost:3000/api/v1/") };
        _client.DefaultRequestHeaders.Add("API_KEY", "a1b2c3d4e5");
    }

    private async Task<int> CreateClientAsync(Client client)
    {
        var content = new StringContent(JsonConvert.SerializeObject(client), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("clients", content);
        response.EnsureSuccessStatusCode(); // Throw if not a success code.
        var createdClient = JsonConvert.DeserializeObject<Client>(await response.Content.ReadAsStringAsync());
        return createdClient.Id;
    }

    [Fact]
    public async Task Delete_Client()
    {
        var client = new Client { Name = "Test Client", Address = "123 Test St", City = "Test City", Zip_code = "12345" };
        int clientId = await CreateClientAsync(client);

        var response = await _client.DeleteAsync($"clients/{clientId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_Client()
    {
        var response = await _client.DeleteAsync("clients/-1"); // Assume this ID does not exist
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Client_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync($"clients/1");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    private async Task<int> CreateShipmentAsync(Shipment shipment)
    {
        var content = new StringContent(JsonConvert.SerializeObject(shipment), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("shipments", content);
        response.EnsureSuccessStatusCode(); // Throw if not a success code.
        var createdShipment = JsonConvert.DeserializeObject<Shipment>(await response.Content.ReadAsStringAsync());
        return createdShipment.Id;
    }

    [Fact]
    public async Task Delete_Shipment_Should()
    {
        var shipment = new Shipment
        {
            Order_Id = 1,
            Source_Id = 1,
            Order_Date = "2024-10-18",
            Request_Date = "2024-10-19",
            Shipment_Date = "2024-10-20",
            Shipment_Type = "Express",
            Shipment_Status = "Shipped",
            Items = new List<ShipmentItem>
            {
                new ShipmentItem { Item_Id = "item1", Amount = 1 }
            }
        };
        int shipmentId = await CreateShipmentAsync(shipment);

        var response = await _client.DeleteAsync($"shipments/{shipmentId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_Shipment()
    {
        var response = await _client.DeleteAsync("shipments/-1"); // Assume this ID does not exist
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Shipment_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync($"shipments/1"); // Assume this shipment has dependent data
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    private async Task<int> CreateItemGroupAsync(ItemGroup itemGroup)
    {
        var content = new StringContent(JsonConvert.SerializeObject(itemGroup), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("itemgroups", content);
        response.EnsureSuccessStatusCode(); // Throw if not a success code.
        var createdItemGroup = JsonConvert.DeserializeObject<ItemGroup>(await response.Content.ReadAsStringAsync());
        return createdItemGroup.Id;
    }

    [Fact]
    public async Task Delete_ItemGroup_Should()
    {
        var itemGroup = new ItemGroup { Name = "Test Item Group", Description = "Description for test group" };
        int itemGroupId = await CreateItemGroupAsync(itemGroup);

        var response = await _client.DeleteAsync($"item_groups/{itemGroupId}");
        Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_ItemGroup()
    {
        var response = await _client.DeleteAsync("item_groups/-1"); // Assume this ID does not exist
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_ItemGroup_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync($"item_groups/1"); // Assume this item group has dependent data
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    private async Task<int> CreateSupplierAsync(Supplier supplier)
    {
        var content = new StringContent(JsonConvert.SerializeObject(supplier), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("suppliers", content);
        response.EnsureSuccessStatusCode();
        var createdSupplier = JsonConvert.DeserializeObject<Supplier>(await response.Content.ReadAsStringAsync());
        return createdSupplier.Id;
    }
    
    [Fact]
    public async Task Delete_Supplier()
    {
        var supplier = new Supplier { Name = "Test Supplier", Address = "123 Supplier St" };
        int supplierId = await CreateSupplierAsync(supplier);
        var response = await _client.DeleteAsync($"suppliers/{supplierId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_Supplier()
    {
        var response = await _client.DeleteAsync("suppliers/-1");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Supplier_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync($"suppliers/1");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }


    private async Task<int> CreateTransferAsync(Transfer transfer)
    {
        var content = new StringContent(JsonConvert.SerializeObject(transfer), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("transfers", content);
        response.EnsureSuccessStatusCode();
        var createdTransfer = JsonConvert.DeserializeObject<Transfer>(await response.Content.ReadAsStringAsync());
        return createdTransfer.Id;
    }

    [Fact]
    public async Task Delete_Transfer()
    {
        var transfer = new Transfer
        {
            Items = new List<TransferItem>
            {
                new TransferItem { Item_Id = "item1" },
                new TransferItem { Item_Id = "item2" }
            },
            Transfer_Status = "Pending"
        };
        int transferId = await CreateTransferAsync(transfer);
        var response = await _client.DeleteAsync($"transfers/{transferId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_Transfer()
    {
        var response = await _client.DeleteAsync("transfers/-1");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Transfer_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync($"transfers/1");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    private async Task<int> CreateWarehouseAsync(Warehouse warehouse)
    {
        var content = new StringContent(JsonConvert.SerializeObject(warehouse), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("warehouses", content);
        response.EnsureSuccessStatusCode();
        var createdWarehouse = JsonConvert.DeserializeObject<Warehouse>(await response.Content.ReadAsStringAsync());
        return createdWarehouse.Id;
    }

    [Fact]
    public async Task Delete_Warehouse()
    {
        var warehouse = new Warehouse { Name = "Test Warehouse", Address = "Test Address" };
        int warehouseId = await CreateWarehouseAsync(warehouse);
        var response = await _client.DeleteAsync($"warehouses/{warehouseId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_Warehouse()
    {
        var response = await _client.DeleteAsync("warehouses/-1");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Warehouse_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync($"warehousess/1");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    private async Task<int> CreateItemTypeAsync(ItemType itemType)
    {
        var content = new StringContent(JsonConvert.SerializeObject(itemType), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("item_types", content);
        response.EnsureSuccessStatusCode(); // Throw if not a success code.
        var createdItemType = JsonConvert.DeserializeObject<ItemType>(await response.Content.ReadAsStringAsync());
        return createdItemType.Id;
    }

    [Fact]
    public async Task Delete_ItemType_Should()
    {   
        var itemType = new ItemType { Name = "Test Item Type", Description = "Description for test type" };
        int itemTypeId = await CreateItemTypeAsync(itemType);

        var response = await _client.DeleteAsync("item_types/1");
        Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_ItemType()
    {
        var response = await _client.DeleteAsync("item_types/-1"); // Assume this ID does not exist
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_ItemType_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync("item_types/1"); // Assume this item type has dependent data
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    private async Task<string> CreateItemAsync(Item item)
    {
        var content = new StringContent(JsonConvert.SerializeObject(item), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("items", content);
        response.EnsureSuccessStatusCode(); // Throw if not a success code.
        var createdItem = JsonConvert.DeserializeObject<Item>(await response.Content.ReadAsStringAsync());
        return createdItem.Uid;
    }

    [Fact]
    public async Task Delete_Item_Should()
    {
        var item = new Item
        {
            Code = "Test Item",
            Description = "Description for test item",
            Short_Description = "Short description",
            Upc_Code = "1234567890",
            Model_Number = "Model 123",
            Commodity_Code = "Commodity 123",
            Item_Line = 1,
            Item_Group = 1,
            Item_Type = 1,
            Unit_Purchase_Quantity = 1,
            Unit_Order_Quantity = 1,
            Pack_Order_Quantity = 1,
            Supplier_Id = 1,
            Supplier_Code = "Supplier 123",
            Supplier_Part_Number = "Part 123"
        };
        string itemUid = await CreateItemAsync(item);

        var response = await _client.DeleteAsync($"items/{itemUid}");
        Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_Item()
    {
        var response = await _client.DeleteAsync("items/-1"); // Assume this ID does not exist
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Item_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync("items/1"); // Assume this item has dependent data
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }


    private async Task<int> CreateOrderAsync(Order order)
    {
        var content = new StringContent(JsonConvert.SerializeObject(order), Encoding.UTF8, "application/json");
        var response = await _client.PostAsync("orders", content);
        response.EnsureSuccessStatusCode(); // Throw if not a success code.
        var createdOrder = JsonConvert.DeserializeObject<Order>(await response.Content.ReadAsStringAsync());
        return createdOrder.Id;
    }

    [Fact]
    public async Task Delete_Order_Should()
    {
        var order = new Order
        {
            Source_Id = 1,
            Order_Date = "2024-10-18",
            Request_Date = "2024-10-19",
            Reference = "Test Order",
            Order_Status = "Scheduled",
            Items = new List<OrderItem>
            {
                new OrderItem { Item_Id = "item1", Amount = 1 }
            }
        };
        int orderId = await CreateOrderAsync(order);

        var response = await _client.DeleteAsync($"orders/{orderId}");
        Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Non_Existent_Order()
    {
        var response = await _client.DeleteAsync("orders/-1"); // Assume this ID does not exist
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Delete_Order_When_Other_Data_Is_Dependent()
    {
        var response = await _client.DeleteAsync("orders/1"); // Assume this order has dependent data
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
}

File: C#api/Tests/test_delete_handler.py
import httpx
import unittest
import random
import string
import json
import os

def generate_uid():
    return ''.join(random.choices(string.ascii_letters + string.digits, k=10))

class ApiDeleteTests(unittest.TestCase):
    def setUp(self):
        self.base_url = "http://127.0.0.1:3000/api/v1/"
        self.client = httpx.Client(headers={"API_KEY": "a1b2c3d4e5"})
        self.data_root = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))), "data").replace(os.sep, "/")

    def GetJsonData(self, model):
        with open(os.path.join(self.data_root, f"{model}.json"), 'r') as file:
            data = json.load(file)
        return data

    def test_delete_client(self):
        client_data = '{"id": 1234567890, "name": "Test Client", "address": "123 Test St", "city": "Test City", "zip_code": "12345"}'
        response = self.client.post(f"{self.base_url}clients", data=client_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(client_data, self.GetJsonData("clients"))
        client_id = 1234567890  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}clients/{client_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(client_data, self.GetJsonData("clients"))

    def test_delete_non_existent_client(self):
        response = self.client.delete(f"{self.base_url}clients/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_shipment(self):
        shipment_data = '''{
            "id": 1234567891,
            "order_id": 1,
            "source_id": 1,
            "order_date": "2024-10-18",
            "request_date": "2024-10-19",
            "shipment_date": "2024-10-20",
            "shipment_type": "Express",
            "shipment_status": "Shipped",
            "items": [{"item_id": "item1", "amount": 1}]
        }'''
        response = self.client.post(f"{self.base_url}shipments", data=shipment_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(shipment_data, self.GetJsonData("shipments"))
        shipment_id = 1234567891  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}shipments/{shipment_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(shipment_data, self.GetJsonData("shipments"))

    def test_delete_non_existent_shipment(self):
        response = self.client.delete(f"{self.base_url}shipments/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_item_group(self):
        item_group_data = '{"id": 1234567892, "name": "Test Item Group", "description": "Description for test group"}'
        response = self.client.post(f"{self.base_url}item_groups", data=item_group_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(item_group_data, self.GetJsonData("item_groups"))
        item_group_id = 1234567892  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}item_groups/{item_group_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(item_group_data, self.GetJsonData("item_groups"))

    def test_delete_non_existent_item_group(self):
        response = self.client.delete(f"{self.base_url}item_groups/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_supplier(self):
        supplier_data = '{"id": 1234567893, "name": "Test Supplier", "address": "123 Supplier St"}'
        response = self.client.post(f"{self.base_url}suppliers", data=supplier_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(supplier_data, self.GetJsonData("suppliers"))
        supplier_id = 1234567893  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}suppliers/{supplier_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(supplier_data, self.GetJsonData("suppliers"))

    def test_delete_non_existent_supplier(self):
        response = self.client.delete(f"{self.base_url}suppliers/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_transfer(self):
        transfer_data = '{"id": 1234567894, "items": [{"item_id": "item1"}, {"item_id": "item2"}], "transfer_status": "Pending"}'
        response = self.client.post(f"{self.base_url}transfers", data=transfer_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(transfer_data, self.GetJsonData("transfers"))
        transfer_id = 1234567894  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}transfers/{transfer_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(transfer_data, self.GetJsonData("transfers"))

    def test_delete_non_existent_transfer(self):
        response = self.client.delete(f"{self.base_url}transfers/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_warehouse(self):
        warehouse_data = '{"id": 1234567895, "name": "Test Warehouse", "address": "Test Address"}'
        response = self.client.post(f"{self.base_url}warehouses", data=warehouse_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(warehouse_data, self.GetJsonData("warehouses"))
        warehouse_id = 1234567895  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}warehouses/{warehouse_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(warehouse_data, self.GetJsonData("warehouses"))

    def test_delete_non_existent_warehouse(self):
        response = self.client.delete(f"{self.base_url}warehouses/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_item_type(self):
        item_type_data = '{"id": 1234567896, "name": "Test Item Type", "description": "Description for test type"}'
        response = self.client.post(f"{self.base_url}item_types", data=item_type_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(item_type_data, self.GetJsonData("item_types"))
        item_type_id = 1234567896  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}item_types/{item_type_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(item_type_data, self.GetJsonData("item_types"))

    def test_delete_non_existent_item_type(self):
        response = self.client.delete(f"{self.base_url}item_types/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_item_line(self):
        item_line_data = '{"id": 1234567897, "name": "Test ItemLine", "description": "Description of the test item line"}'
        response = self.client.post(f"{self.base_url}item_lines", data=item_line_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(item_line_data, self.GetJsonData("item_lines"))
        item_line_id = 1234567897  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}item_lines/{item_line_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(item_line_data, self.GetJsonData("item_lines"))

    def test_delete_non_existent_item_line(self):
        response = self.client.delete(f"{self.base_url}item_lines/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_item(self):
        item_data = '''{
            "uid": "hgdfhjnfJFBDH",
            "code": "Test Item",
            "description": "Description for test item",
            "short_description": "Short description",
            "upc_code": "1234567890",
            "model_number": "Model 123",
            "commodity_code": "Commodity 123",
            "item_line": 1,
            "item_group": 1,
            "item_type": 1,
            "unit_purchase_quantity": 1,
            "unit_order_quantity": 1,
            "pack_order_quantity": 1,
            "supplier_id": 1,
            "supplier_code": "Supplier 123",
            "supplier_part_number": "Part 123"
        }'''
        response = self.client.post(f"{self.base_url}items", data=item_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(item_data, self.GetJsonData("items"))
        item_uid = "hgdfhjnfJFBDH"  # Replace with the actual UID or another identifier if needed

        response = self.client.delete(f"{self.base_url}items/{item_uid}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(item_data, self.GetJsonData("items"))

    def test_delete_non_existent_item(self):
        response = self.client.delete(f"{self.base_url}items/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_order(self):
        order_data = '''{
            "id": 1234567898,
            "source_id": 1,
            "order_date": "2024-10-18",
            "request_date": "2024-10-19",
            "reference": "Test Order",
            "order_status": "Scheduled",
            "items": [{"item_id": "item1", "amount": 1}]
        }'''
        response = self.client.post(f"{self.base_url}orders", data=order_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(order_data, self.GetJsonData("orders"))
        order_id = 1234567898  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}orders/{order_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(order_data, self.GetJsonData("orders"))

    def test_delete_non_existent_order(self):
        response = self.client.delete(f"{self.base_url}orders/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_inventory(self):
        inventory_data = '''{
            "id": 1234567899,
            "item_id": "P000TEST",
            "description": "Test Inventory",
            "locations": [1, 2],
            "total_on_hand": 100
        }'''
        response = self.client.post(f"{self.base_url}inventories", data=inventory_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(inventory_data, self.GetJsonData("locations"))
        inventory_id = 1234567899  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}inventories/{inventory_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(inventory_data, self.GetJsonData("inventories"))

    def test_delete_non_existent_inventory(self):
        response = self.client.delete(f"{self.base_url}inventories/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def test_delete_location(self):
        location_data = '{"id": 156738, "warehouse_id": 1, "code": "A.1.0", "name": "Row: A, Rack: 1, Shelf: 0", "created_at": "1992-05-15 03:21:32", "updated_at": "1992-05-15 03:21:32"}'
        response = self.client.post(f"{self.base_url}locations", data=location_data)
        self.assertEqual(response.status_code, httpx.codes.CREATED)
        self.assertIn(location_data, self.GetJsonData("locations"))
        location_id = 156738  # Directly using the ID from the data

        response = self.client.delete(f"{self.base_url}locations/{location_id}")
        self.assertEqual(response.status_code, httpx.codes.OK)
        self.assertNotIn(location_data, self.GetJsonData("locations"))

    def test_delete_non_existent_location(self):
        response = self.client.delete(f"{self.base_url}locations/-1")
        self.assertEqual(response.status_code, httpx.codes.NOT_FOUND)

    def tearDown(self):
        self.client.close()

if __name__ == "__main__":
    unittest.main()
